rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/model/SessionRepresentation.java

```diff
@@ -79,5 +80,18 @@ public Map<String, String> getClients() {
     public void setClients(Map<String, String> clients) {
         this.clients = clients;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
Joshua Bloch recommends to use `instanceof` instead of `getClass()` here.  That way, this method will still work if the class hierarchy changes.  (I always pull out my copy of Effective Java whenever I have to do this).

Reply:
Well if it's not the same class it's not the same imo

Reply:
A distinct session is one that has the same start, userId, and type.  Let's say someone decides we need FooSessionRepresentation and BarSessionRepresentation that both extend SessionRepresentation:

```
FooSessionRepresentation extends SessionRepresentation {
}

BarSessionRepresentation extends SessionRepresentation {
}
```

Now we would be back to having the same problem as in the issue.  There would be a possibility of duplicates with the same start, userId, and type.  They would differ only by class instance, but are really two representations of the same session. So how do you fix this problem?  Just override and call super();

```
FooSessionRepresentation extends SessionRepresentation {
   @Override
    public boolean equals(Object o) {
         return super.equals(o);
    }

   @Override
    public int hashCode() {
        return super.hashCode();
    }
}
```

But with `getClass()`, `super.equals(o)` doesn't work.  `FooSessionRepresentation` and `BarSessionRepresentation` would not be considered equal, even though they are just two representations of the same session.  Nobody would notice the problem and we've just reintroduced the original bug.  With `instanceof`, it does work because both are an instance of `SessionRepresentation`.

It's a pretty esoteric point.  But this is why Bloch recommends `instanceof`.  I think `instanceof` is slightly more correct.